RecoBTag/ONNXRuntime/plugins/ParticleNetSonicJetTagsProducer.cc

```diff
@@ -128,10 +129,39 @@ void ParticleNetSonicJetTagsProducer::acquire(edm::Event const &iEvent, edm::Eve
   edm::Handle<TagInfoCollection> tag_infos;
   iEvent.getByToken(src_, tag_infos);
   client_->setBatchSize(tag_infos->size());
+  skippedInference_ = false;
   if (!tag_infos->empty()) {
+
+    unsigned int maxParticles = 0;
+    unsigned int maxVertices = 0;
+    for (unsigned jet_n = 0; jet_n < tag_infos->size(); ++jet_n) {
+      if( ((*tag_infos)[jet_n]).features().get("pfcand_etarel").size() > maxParticles) maxParticles = ((*tag_infos)[jet_n]).features().get("pfcand_etarel").size();
+      if( ((*tag_infos)[jet_n]).features().get("sv_etarel").size() > maxVertices) maxVertices = ((*tag_infos)[jet_n]).features().get("sv_etarel").size();
+    }
+    if(maxParticles == 0 && maxVertices == 0){
+      client_->setBatchSize(0);
+      skippedInference_ = true;
+      return;
+    }
+    unsigned int minPartFromJSON = prep_info_map_.at(input_names_[0]).min_length;
+    unsigned int maxPartFromJSON = prep_info_map_.at(input_names_[0]).max_length;
+    unsigned int minVertFromJSON = prep_info_map_.at(input_names_[3]).min_length;
+    unsigned int maxVertFromJSON = prep_info_map_.at(input_names_[3]).max_length;
+    maxParticles = std::clamp(maxParticles, minPartFromJSON, maxPartFromJSON);
+    maxVertices = std::clamp(maxVertices, minVertFromJSON, maxVertFromJSON);
+
     for (unsigned igroup = 0; igroup < input_names_.size(); ++igroup) {
       const auto &group_name = input_names_[igroup];
       auto &input = iInput.at(group_name);
+      unsigned target;
+      if(igroup < 3){
```

Comment:
3 seems like a magic number. can you make it into a named constant so it's clearer and easier to change consistently if necessary?

Reply:
I changed this to refer to a number of particle groups, which is set globally (all of the particle net models have 3 particle groups and 3 vector groups in their inputs)